packages/flutter/lib/src/material/drop_down.dart

```diff
@@ -479,19 +477,24 @@ class _DropDownButtonState<T> extends State<DropDownButton<T>> {
     _currentRoute?.style = style;
     Widget result = new DefaultTextStyle(
       style: style,
-      child: new Row(
-        mainAxisAlignment: MainAxisAlignment.collapse,
-        children: <Widget>[
-          // We use an IndexedStack to make sure we have enough width to show any
-          // possible item as the selected item without changing size.
-          new IndexedStack(
-            key: _itemKey,
-            index: _selectedIndex,
-            alignment: FractionalOffset.centerLeft,
-            children: config.items
-          ),
-          new Icon(icon: Icons.arrow_drop_down, size: config.iconSize)
-        ]
+      child: new LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          return new Row(
+            mainAxisAlignment: constraints.hasTightWidth
```

Comment:
I think it's weird that this means we get radically different behaviour for constraints 189..190 vs 190..190.